dace/codegen/targets/cpu.py

```diff
@@ -148,11 +152,10 @@ def generate_node(self, sdfg, dfg, state_id, node, function_stream,
 
         gen(sdfg, dfg, state_id, node, function_stream, callsite_stream)
 
+        self._locals.clear_scope(self._ldepth + 1)
         # Mark node as "generated"
         self._generated_nodes.add(node)
 
-        self._locals.clear_scope(self._ldepth + 1)
-
     def allocate_array(self, sdfg, dfg, state_id, node, function_stream,
```

Comment:
That might not be a good idea

Reply:
Thanks for spotting it. Some wrong editing!